Gracie/dashboard lending

artsygracie17 opened the pull request.

Commit:
Update StartMyCollection design to new Zeplin version. Fix transition from welcome screen to AddComic. Create Dashboard and Dashboard navigation controller. Link profile to navigation smiley icon in dashboard. Decode token in separate AuthenticationResponse model.

Add setNavBarBackground and setNavBarTitle to LucienTheme to be accessed and customized throughout application whenever nav bar needs to be created

Restructure UINavigationController. Navigation flow from Dashboard to Profile working

Authorization token now set properly. Authenticate user everytime at sign in even if already signed in

Commit changes; in middle of getting getCurrentUser endpoint working

getCurrentUser working. Add AuthenticationResponse model for authenticate user and use User model for getCurrentUser only.

Add sample image

Update StartMyCollection design, components and constraints to match new Zeplin design; include linking to Dashboard

Update StartMyCollection xib design to Zeplin state

Add UILabel extension for letter spacing; StartMyCollection welcome screen UI elements updated

Edit constraints in home screen for aspect ratios conformity on iPhone touch

Tweak StartMyCollection screen for device demo

Fix transition from welcome screen to AddCommic screen after rebase. Glitches fixed; add hard-coded scrolling on Dashboard xib for demo

Change color of hard coded scrolling views and direction lock

Delete GoogleAuthentication.swift

Revert back Guled AddComic changes; still not working

AddComicController working

Lucien Profile title glitch fixed

Committing changes to rebase on develop for Jenkins stuff

Commit:
Create Dashboard and Dashboard navigation controller. Link profile to navigation smiley icon in dashboard. Decode token in separate AuthenticationResponse model.

Add setNavBarBackground and setNavBarTitle to LucienTheme to be accessed and customized throughout application whenever nav bar needs to be created

Restructure UINavigationController. Navigation flow from Dashboard to Profile working

Authorization token now set properly. Authenticate user everytime at sign in even if already signed in

Commit changes; in middle of getting getCurrentUser endpoint working

getCurrentUser working. Add AuthenticationResponse model for authenticate user and use User model for getCurrentUser only.

Add sample image

Update StartMyCollection design, components and constraints to match new Zeplin design; include linking to Dashboard

Update StartMyCollection xib design to Zeplin state

Add UILabel extension for letter spacing; StartMyCollection welcome screen UI elements updated

Edit constraints in home screen for aspect ratios conformity on iPhone touch

Tweak StartMyCollection screen for device demo

Fix transition from welcome screen to AddCommic screen after rebase. Glitches fixed; add hard-coded scrolling on Dashboard xib for demo

Change color of hard coded scrolling views and direction lock

Delete GoogleAuthentication.swift

Revert back Guled AddComic changes; still not working

AddComicController working

Lucien Profile title glitch fixed

Committing changes to rebase on develop for Jenkins stuff

UICollectionView in separate xib file added. Add separate xib for UICollectionViewCell then load xib to main UICollectionView view controller

Change size of LendingCollectionViewCell but not showing up properly. Last commit before breaking PR/branch to smaller parts

Trim extra lines

Remove GoogleAuthentication.swift

Update fonts where they were not updated previously. Change app delegate root controller back to RootViewController

Remove extra whitespace

PR changes with exception of a few

hasCollection String to Bool working

Commit:
Add UserResponse model to conform User to borrower and owner. Add DashboardResponse in same fashion and add lending decode to Dashboard model. UICollectionView in separate xib file added. Add separate xib for UICollectionViewCell then load xib to main UICollectionView view controller

Change size of LendingCollectionViewCell but not showing up properly. Last commit before breaking PR/branch to smaller parts

Trim extra lines

Remove GoogleAuthentication.swift

Update fonts where they were not updated previously. Change app delegate root controller back to RootViewController

Remove extra whitespace

PR changes with exception of a few

Add UserResponse model to conform User to borrower and owner. Add DashboardResponse in same fashion

Add lending decode to Dashboard model

commiting changes

Commit:
Adjust Dashboard Model flow so getDashboard works and retrieves lending information.

Merge conflicts on rebase

Commit:
Move UICollectionView delegate, datasource and methods to main DashboardViewController. Redirect sign in back to application on older OS versions

Move UICollectionView delegate, datasource and methods to main DashboardViewController

Commit:
Design Comic cell and make data flow into cells dynamically. Comic title cell showing up after UICollectionView restructuring

Data showing up on UICollectionViews in Dashboard lending

Design comic cell with fonts, font size and constraints and display on Dashboard.

Committing changes

IntrepidBuildServer commented on lucien-ios/Dashboard/LendingCollectionViewController.swift:
```diff
@@ -0,0 +1,67 @@
+
+    func setLayout() {
+        let collectionViewLayout = collectionView.collectionViewLayout as? UICollectionViewFlowLayout
+        let sectionInset = UIEdgeInsetsMake(0.0, 24.0, 0.0, 0.0)
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Legacy Constructor Violation: Swift constructors are preferred over legacy convenience functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Alegacy_constructor)

mayasaxena replied:
Do you know what this error message means?

IntrepidBuildServer commented on lucien-ios/Dashboard/LendingCollectionViewController.swift:
```diff
@@ -0,0 +1,67 @@
+    }
+
+    func collectionView(_ collectionView: UICollectionView, layout
+        collectionViewLayout: UICollectionViewLayout,
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Vertical Parameter Alignment Violation: Function parameters should be aligned vertically if they're in multiple lines in a declaration. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Avertical_parameter_alignment)

IntrepidBuildServer commented on lucien-ios/Dashboard/DashboardCollectionView.swift:
```diff
@@ -0,0 +1,71 @@
+    }
+
+    func collectionView(_ collectionView: UICollectionView, layout
+        collectionViewLayout: UICollectionViewLayout,
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Vertical Parameter Alignment Violation: Function parameters should be aligned vertically if they're in multiple lines in a declaration. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Avertical_parameter_alignment)

IntrepidBuildServer commented:
SonarQube analysis reported 4 issues
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

IntrepidBuildServer commented on lucien-ios/Dashboard/DashboardCollectionView.swift:
```diff
@@ -0,0 +1,71 @@
+
+    func setLayout() {
+        let collectionViewLayout = self.collectionViewLayout as? UICollectionViewFlowLayout
+        let sectionInset = UIEdgeInsetsMake(0.0, 24.0, 0.0, 0.0)
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Legacy Constructor Violation: Swift constructors are preferred over legacy convenience functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Alegacy_constructor)

IntrepidBuildServer commented:
<!--
  0 Errors
  1 Warning: Please provide a summary in th...
  2 Messages: Executed 2 tests, with 0 failu...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the Pull Request description.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.410 (0.415) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 1 test, with 0 failures (0 unexpected) in 7.869 (7.871) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

IntrepidBuildServer commented:
<!--
  0 Errors
  1 Warning: Please provide a summary in th...
  2 Messages: Executed 2 tests, with 0 failu...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the Pull Request description.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.356 (0.358) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 1 test, with 0 failures (0 unexpected) in 6.240 (6.242) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

Commit:
More CollectionViewCell changes

mayasaxena commented on lucien-ios/AppDelegate.swift:
```diff
@@ -23,11 +23,18 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
+
+    func application(_ app: UIApplication, open url: URL, options: [UIApplicationOpenURLOptionsKey : Any] = [:]) -> Bool {
+
+        let handled = GIDSignIn.sharedInstance().handle(url, sourceApplication:options[UIApplicationOpenURLOptionsKey.sourceApplication] as? String, annotation: [:])
```
You can just return this line, rather than storing it and returning it.

mayasaxena commented on lucien-ios/Dashboard/DashboardCollectionView.swift:
```diff
@@ -0,0 +1,71 @@
+
+import UIKit
+
+class DashboardCollectionView: UICollectionView, UICollectionViewDataSource, UICollectionViewDelegate {
```
Should be final

mayasaxena commented on lucien-ios/Dashboard/DashboardCollectionView.swift:
```diff
@@ -0,0 +1,71 @@
+        self.dataSource = self
+        self.delegate = self
+        let nib = UINib(nibName: "CollectionViewCell", bundle: nil)
+        self.register(nib, forCellWithReuseIdentifier: reuseIdentifier)
```
Remove `self`

IntrepidBuildServer commented:
<!--
  1 Error: fatal error: file ‘/Users/jenk...
  1 Warning: Please provide a summary in th...
  2 Messages: Executed 2 tests, with 0 failu...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">fatal error: file ‘/Users/jenkins/Jenkins/workspace/s-pipeline_lucien-ios_PR-17-OWD2SBTDO5QNYJU26PKWTLOAZHXLI2SNBF6MUEWFJC6EOY6SY7MQ/Pods/Google/Headers/module.modulemap’ has been modified since the precompiled header ‘/Users/jenkins/Jenkins/workspace/s-pipeline_lucien-ios_PR-17-OWD2SBTDO5QNYJU26PKWTLOAZHXLI2SNBF6MUEWFJC6EOY6SY7MQ/DerivedData/Build/Intermediates.noindex/PrecompiledHeaders/lucien-ios-Bridging-Header-swift_1J727JDM20TYY-clang_2O9YIEWVALUCS.pch’ was built</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the Pull Request description.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.481 (0.483) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 1 test, with 0 failures (0 unexpected) in 13.110 (13.112) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

mayasaxena commented on lucien-ios/Dashboard/DashboardCollectionViewModel.swift:
```diff
@@ -0,0 +1,46 @@
+
+    func getComicCount() -> Int {
+        guard let count = lendedComics?.count else { return 0 }
+        return count
```
You can turn this into a one-liner `return lendedComics.count ?? 0`

mayasaxena commented on lucien-ios/Dashboard/DashboardCollectionViewModel.swift:
```diff
@@ -0,0 +1,46 @@
+    private var lendedComics = DashboardViewModel.dashboardData?.lending
+    private var borrowedComics = DashboardViewModel.dashboardData?.borrowing
+
+    func getComicCount() -> Int {
```
If what you're returning is just a variable belonging to a private property in the view model, you should expose it in the form of a variable, not a function

mayasaxena commented on lucien-ios/Dashboard/DashboardCollectionViewModel.swift:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    func comicTitle(forIndex index: Int) -> String {
+        return lendedComics![index].comicTitle
```
Don't force unwrap

mayasaxena commented on lucien-ios/Dashboard/DashboardCollectionViewModel.swift:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    func comicDueDate(forIndex index: Int) -> Date {
+        guard let returnDate = lendedComics![index].returnDate else { return Date()}
```
Don't force unwrap

mayasaxena commented on lucien-ios/Models/Dashboard.swift:
```diff
@@ -9,19 +9,18 @@
     var myCollectionCount: Int
+    var lendingCount: Int
+    var borrowingCount: Int
+    var lending: [Comic]
```
This naming is not very descriptive. What is being lent? What is being borrowed?

mayasaxena commented on lucien-ios/Models/MyCollection.swift:
```diff
@@ -0,0 +1,13 @@
+
+import Foundation
+
+struct MyCollection: Decodable {
```
If you were just using this for testing, remove it before putting up your PR

mayasaxena commented on lucien-ios/Models/UserResponse.swift:
```diff
@@ -0,0 +1,13 @@
+//
+//  NestedUser.swift
```
Update the comment to match the file name

mayasaxena commented on lucien-ios/Dashboard/DashboardViewModel.swift:
```diff
@@ -10,15 +10,14 @@ import Foundation
-    static let lendingCellTitles = ["Princess", "Gracie", "Kittens"]
     let lucienAPIClient = LucienAPIClient()
-    var dashboardData: Dashboard?
+    static var dashboardData: Dashboard?
```
Why is this static and why are you exposing it?

mayasaxena commented on lucien-ios/Dashboard/LendingCollectionViewCell.swift:
```diff
@@ -0,0 +1,13 @@
+import UIKit
+
+class LendingCollectionViewCell: UICollectionViewCell {
+    @IBOutlet weak var lendedComicTitle: UILabel!
```
Labels should have the word label at the end of their names. `lendedComicTitle` looks like it should just be a String or other data type

mayasaxena replied:
This should be private

mayasaxena commented on lucien-ios/Dashboard/LendingCollectionViewCell.swift:
```diff
@@ -0,0 +1,13 @@
+
+import UIKit
+
+class LendingCollectionViewCell: UICollectionViewCell {
```
Should be final

mayasaxena commented on lucien-ios/Dashboard/LendingCollectionViewController.swift:
```diff
@@ -0,0 +1,67 @@
+
+final class LendingCollectionViewController: UIViewController, UICollectionViewDataSource, UICollectionViewDelegate, UICollectionViewDelegateFlowLayout {
+
+    @IBOutlet weak var collectionView: UICollectionView!
```
should be private

mayasaxena commented on lucien-ios/Dashboard/LendingCollectionViewController.swift:
```diff
@@ -0,0 +1,67 @@
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.collectionView.register(UINib(nibName:"LendingCollectionViewCell", bundle: nil), forCellWithReuseIdentifier: reuseIdentifier)
```
Remove all uses of `self`

mayasaxena replied:
Rather than hardcoding in the nib name, use `String(describing: LendingCollectionViewCell.self)`. The function gets the name of the class dynamically so if you ever need to change the cell class things will stay in sync

mayasaxena commented on lucien-ios/Dashboard/LendingCollectionViewController.swift:
```diff
@@ -0,0 +1,67 @@
+        collectionViewLayout?.sectionInset = sectionInset
+    }
+
+    func numberOfSections(in collectionView: UICollectionView) -> Int { return 1 }
```
Function definitions should never be one-liners. Also, 1 is the default value returned so you don't actually need to implement this

mayasaxena commented on lucien-ios/Dashboard/LendingCollectionViewController.swift:
```diff
@@ -0,0 +1,67 @@
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: reuseIdentifier, for: indexPath)
+        guard let lendingCell = cell as? LendingCollectionViewCell else { return cell}
+//        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: reuseIdentifier, for: indexPath) as! LendingCollectionViewCell else { return }
```
Remove commented-out code

mayasaxena commented on lucien-ios/Dashboard/LendingCollectionViewController.swift:
```diff
@@ -0,0 +1,67 @@
+        return lendingCell
+    }
+
+    func configure(cell: LendingCollectionViewCell, lendedComicTitle: String) {
```
This function should be defined in the `LendingCollectionViewCell` class itself so you can keep the UI elements within it private

mayasaxena commented on lucien-ios/Extensions/UIView+Extensions.swift:
```diff
@@ -0,0 +1,19 @@
+import UIKit
+
+extension UIView {
+    class func loadFromNibNamed(nibNamed: String, bundle: Bundle? = nil) -> UIView? {
```
Are you using this anywhere? If not, remove it

mayasaxena commented on lucien-ios/Extensions/UIViewController+Extensions.swift:
```diff
@@ -0,0 +1,25 @@
+
+import Foundation
+
+extension UIViewController {
```
Remove empty extension

mayasaxena commented on lucien-ios/Extensions/UIViewController+Extensions.swift:
```diff
@@ -0,0 +1,25 @@
+
+}
+
+extension UILabel {
```
This shouldn't be in this file, it's an extension on `UILabel` so it should be in a file named `UILabel+Extensions`

mayasaxena commented on lucien-ios/LoginScreen/RootViewController.swift:
```diff
@@ -64,20 +65,19 @@ final class RootViewController: UIViewController, GIDSignInDelegate, GIDSignInUI
-        let dashboardNavigationController = UINavigationController(rootViewController: dashboardViewController)
-        present(dashboardNavigationController, animated: true, completion: nil)
+        dashboardViewModel.getDashboard {
+            guard let dashboardData = DashboardViewModel.dashboardData else { return }
```
This shouldn't be happening in the view controller - exposing the `dashboardData` property defeats the purpose of MVVM

mayasaxena reviewed (changes requested):
This PR is closer to a reasonable size, but it's still a little large. Try breaking up your work into smaller chunks and putting up a PR whenever you start something new. It's okay to have more than one PR up at a time as long as they don't cover the exact same area

Commit:
Refactor dashboard collection flow to allow for lending and borrowing. Adjust cell constraints to fix corner radius bug

Commit:
Add profile picture to Profile page; add AddBook button to right of navigation bar.

Make PR changes except for static var in view model. Comments everywhere for now because viewmodel passing in progress

artsygracie17 commented on lucien-ios/Dashboard/LendingCollectionViewController.swift:
```diff
@@ -0,0 +1,67 @@
+
+    func setLayout() {
+        let collectionViewLayout = collectionView.collectionViewLayout as? UICollectionViewFlowLayout
+        let sectionInset = UIEdgeInsetsMake(0.0, 24.0, 0.0, 0.0)
```
I think it means I used the old obj C way of setting insets -- xcode automatically deletes the parameter descriptions top, left, right, bottom so I input the numbers as is; I'll fix this!

IntrepidBuildServer commented:
<!--
  0 Errors
  1 Warning: Please provide a summary in th...
  2 Messages: Executed 2 tests, with 0 failu...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the Pull Request description.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.373 (0.375) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 1 test, with 0 failures (0 unexpected) in 64.145 (64.149) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

IntrepidBuildServer commented:
<!--
  0 Errors
  1 Warning: Please provide a summary in th...
  2 Messages: Executed 2 tests, with 0 failu...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the Pull Request description.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.412 (0.414) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 1 test, with 0 failures (0 unexpected) in 6.130 (6.131) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

mayasaxena commented on lucien-ios/Dashboard/DashboardCollectionView.swift:
```diff
@@ -0,0 +1,78 @@
+
+class DashboardCollectionView: UICollectionView, UICollectionViewDataSource, UICollectionViewDelegate {
+
+//    var viewModel: DashboardCollectionViewModel!
```
Remove commented-out code

Commit:
Fix structural defects and remove static variable access. PR changes that wer not already made since last push

Remove LendingCollection files -- these should not be here

Remove UIViewController extension and MyCollection Model

Remove files properly

IntrepidBuildServer commented:
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

IntrepidBuildServer commented:
<!--
  1 Error: fatal error: file ‘/Users/jenk...
  3 Warnings: Please provide a summary in th...
  2 Messages: Executed 2 tests, with 0 failu...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">fatal error: file ‘/Users/jenkins/Jenkins/workspace/s-pipeline_lucien-ios_PR-17-OWD2SBTDO5QNYJU26PKWTLOAZHXLI2SNBF6MUEWFJC6EOY6SY7MQ/Pods/Google/Headers/module.modulemap’ has been modified since the precompiled header ‘/Users/jenkins/Jenkins/workspace/s-pipeline_lucien-ios_PR-17-OWD2SBTDO5QNYJU26PKWTLOAZHXLI2SNBF6MUEWFJC6EOY6SY7MQ/DerivedData/Build/Intermediates.noindex/PrecompiledHeaders/lucien-ios-Bridging-Header-swift_275PJW9VA61OG-clang_2O9YIEWVALUCS.pch’ was built</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the Pull Request description.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/IntrepidPursuits/lucien-ios/blob/1d9e8993dbfe404c3ede26373acea09de2f5f8c4/DashboardCollectionViewModel.swift#L34">DashboardCollectionViewModel.swift#L34</a> - <strong><a href="https://github.com/IntrepidPursuits/lucien-ios/blob/1d9e8993dbfe404c3ede26373acea09de2f5f8c4/lucien-ios/Dashboard/DashboardCollectionViewModel.swift#L34">lucien-ios/Dashboard/DashboardCollectionViewModel.swift#L34</a></strong>: left side of nil coalescing operator ‘??’ has non-optional type ‘String’, so the right side is never used  <br /><code>
        let firstName = dashboardComics[index].dashboardUser.firstName ?? ""
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/IntrepidPursuits/lucien-ios/blob/1d9e8993dbfe404c3ede26373acea09de2f5f8c4/DashboardCollectionViewModel.swift#L35">DashboardCollectionViewModel.swift#L35</a> - <strong><a href="https://github.com/IntrepidPursuits/lucien-ios/blob/1d9e8993dbfe404c3ede26373acea09de2f5f8c4/lucien-ios/Dashboard/DashboardCollectionViewModel.swift#L35">lucien-ios/Dashboard/DashboardCollectionViewModel.swift#L35</a></strong>: left side of nil coalescing operator ‘??’ has non-optional type ‘String’, so the right side is never used  <br /><code>
        let lastName = dashboardComics[index].dashboardUser.lastName ?? ""
</code></td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.380 (0.384) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 1 test, with 0 failures (0 unexpected) in 6.292 (6.294) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

IntrepidBuildServer commented:
<!--
  1 Error: fatal error: file ‘/Users/jenk...
  3 Warnings: Please provide a summary in th...
  2 Messages: Executed 2 tests, with 0 failu...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">fatal error: file ‘/Users/jenkins/Jenkins/workspace/s-pipeline_lucien-ios_PR-17-OWD2SBTDO5QNYJU26PKWTLOAZHXLI2SNBF6MUEWFJC6EOY6SY7MQ/Pods/Google/Headers/module.modulemap’ has been modified since the precompiled header ‘/Users/jenkins/Jenkins/workspace/s-pipeline_lucien-ios_PR-17-OWD2SBTDO5QNYJU26PKWTLOAZHXLI2SNBF6MUEWFJC6EOY6SY7MQ/DerivedData/Build/Intermediates.noindex/PrecompiledHeaders/lucien-ios-Bridging-Header-swift_275PJW9VA61OG-clang_2O9YIEWVALUCS.pch’ was built</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the Pull Request description.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/IntrepidPursuits/lucien-ios/blob/e44a0bd282454dc81170556ed01819ef8e06d44c/DashboardCollectionViewModel.swift#L34">DashboardCollectionViewModel.swift#L34</a> - <strong><a href="https://github.com/IntrepidPursuits/lucien-ios/blob/e44a0bd282454dc81170556ed01819ef8e06d44c/lucien-ios/Dashboard/DashboardCollectionViewModel.swift#L34">lucien-ios/Dashboard/DashboardCollectionViewModel.swift#L34</a></strong>: left side of nil coalescing operator ‘??’ has non-optional type ‘String’, so the right side is never used  <br /><code>
        let firstName = dashboardComics[index].dashboardUser.firstName ?? ""
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/IntrepidPursuits/lucien-ios/blob/e44a0bd282454dc81170556ed01819ef8e06d44c/DashboardCollectionViewModel.swift#L35">DashboardCollectionViewModel.swift#L35</a> - <strong><a href="https://github.com/IntrepidPursuits/lucien-ios/blob/e44a0bd282454dc81170556ed01819ef8e06d44c/lucien-ios/Dashboard/DashboardCollectionViewModel.swift#L35">lucien-ios/Dashboard/DashboardCollectionViewModel.swift#L35</a></strong>: left side of nil coalescing operator ‘??’ has non-optional type ‘String’, so the right side is never used  <br /><code>
        let lastName = dashboardComics[index].dashboardUser.lastName ?? ""
</code></td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.371 (0.373) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 1 test, with 0 failures (0 unexpected) in 6.397 (6.399) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

Commit:
Fix minor styling

IntrepidBuildServer commented:
<!--
  0 Errors
  3 Warnings: Please provide a summary in th...
  2 Messages: Executed 2 tests, with 0 failu...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the Pull Request description.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/IntrepidPursuits/lucien-ios/blob/3ac2b2f8d8ab14f868278bbfeec407613855f1e0/DashboardCollectionViewModel.swift#L34">DashboardCollectionViewModel.swift#L34</a> - <strong><a href="https://github.com/IntrepidPursuits/lucien-ios/blob/3ac2b2f8d8ab14f868278bbfeec407613855f1e0/lucien-ios/Dashboard/DashboardCollectionViewModel.swift#L34">lucien-ios/Dashboard/DashboardCollectionViewModel.swift#L34</a></strong>: left side of nil coalescing operator ‘??’ has non-optional type ‘String’, so the right side is never used  <br /><code>
        let firstName = dashboardComics[index].dashboardUser.firstName ?? ""
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/IntrepidPursuits/lucien-ios/blob/3ac2b2f8d8ab14f868278bbfeec407613855f1e0/DashboardCollectionViewModel.swift#L35">DashboardCollectionViewModel.swift#L35</a> - <strong><a href="https://github.com/IntrepidPursuits/lucien-ios/blob/3ac2b2f8d8ab14f868278bbfeec407613855f1e0/lucien-ios/Dashboard/DashboardCollectionViewModel.swift#L35">lucien-ios/Dashboard/DashboardCollectionViewModel.swift#L35</a></strong>: left side of nil coalescing operator ‘??’ has non-optional type ‘String’, so the right side is never used  <br /><code>
        let lastName = dashboardComics[index].dashboardUser.lastName ?? ""
</code></td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.485 (0.487) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 1 test, with 0 failures (0 unexpected) in 8.015 (8.017) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

IntrepidBuildServer commented:
<!--
  1 Error: fatal error: file ‘/Users/jenk...
  1 Warning: Please provide a summary in th...
  2 Messages: Executed 2 tests, with 0 failu...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">fatal error: file ‘/Users/jenkins/Jenkins/workspace/s-pipeline_lucien-ios_PR-17-OWD2SBTDO5QNYJU26PKWTLOAZHXLI2SNBF6MUEWFJC6EOY6SY7MQ/Pods/Google/Headers/module.modulemap’ has been modified since the precompiled header ‘/Users/jenkins/Jenkins/workspace/s-pipeline_lucien-ios_PR-17-OWD2SBTDO5QNYJU26PKWTLOAZHXLI2SNBF6MUEWFJC6EOY6SY7MQ/DerivedData/Build/Intermediates.noindex/PrecompiledHeaders/lucien-ios-Bridging-Header-swift_275PJW9VA61OG-clang_2O9YIEWVALUCS.pch’ was built</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the Pull Request description.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.371 (0.373) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 1 test, with 0 failures (0 unexpected) in 7.456 (7.458) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

Commit:
Add kitty image if no comic photo in db; will change this to default cover

Commit:
Add kitty image if no comic photo in db; will change this to default cover

IntrepidBuildServer commented on lucien-ios/Models/DashboardComic.swift:
```diff
@@ -0,0 +1,68 @@
+        let issueNumber = try? container.decode(String.self, forKey: .issueNumber)
+        let publisher = try? container.decode(String.self, forKey: .publisher)
+        let releaseDate = try? container.decode(Date.self, forKey: .releaseDate)
+        let comicPhotoURL = try? container.decode(String.self, forKey: .comicPhotoURL) 
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Trailing Whitespace Violation: Lines should not have trailing whitespace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Atrailing_whitespace)

IntrepidBuildServer commented:
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

IntrepidBuildServer commented:
<!--
  1 Error: fatal error: file ‘/Users/jenk...
  1 Warning: Please provide a summary in th...
  2 Messages: Executed 2 tests, with 0 failu...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">fatal error: file ‘/Users/jenkins/Jenkins/workspace/s-pipeline_lucien-ios_PR-17-OWD2SBTDO5QNYJU26PKWTLOAZHXLI2SNBF6MUEWFJC6EOY6SY7MQ/Pods/Google/Headers/module.modulemap’ has been modified since the precompiled header ‘/Users/jenkins/Jenkins/workspace/s-pipeline_lucien-ios_PR-17-OWD2SBTDO5QNYJU26PKWTLOAZHXLI2SNBF6MUEWFJC6EOY6SY7MQ/DerivedData/Build/Intermediates.noindex/PrecompiledHeaders/lucien-ios-Bridging-Header-swift_275PJW9VA61OG-clang_2O9YIEWVALUCS.pch’ was built</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the Pull Request description.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.361 (0.363) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 1 test, with 0 failures (0 unexpected) in 11.880 (11.882) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

mayasaxena commented on lucien-ios/Dashboard/DashboardCollectionView.swift:
```diff
@@ -0,0 +1,78 @@
+
+    override init(frame: CGRect, collectionViewLayout layout: UICollectionViewLayout) {
+        super.init(frame: frame, collectionViewLayout: layout)
+        dataSource = self
```
Pull this and the following lines out into a function that you call in both inits so that they're sure to do the same thing

mayasaxena commented on lucien-ios/Dashboard/DashboardCollectionViewModel.swift:
```diff
@@ -0,0 +1,43 @@
+
+final class DashboardCollectionViewModel {
+
+    var dashboardComics: [DashboardComic]
```
This can be a `let` and should be private

mayasaxena commented on lucien-ios/Dashboard/DashboardCollectionViewModel.swift:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    func comicImageURL(forIndex index: Int) -> String {
+        guard let url = dashboardComics[index].dashboardComic.comicPhotoURL else { return "https://www.petsworld.in/blog/wp-content/uploads/2014/09/adorable-cat.jpg" }
```
Is that URL actually what you want to return if the image URL is nil?

artsygracie17 replied:
Originally it was just a placeholder for the actual imageView that will replace the image if a comic book doesn't have a cover, but I realize I can be setting this in the `setImage` of the collection view cell and that this is the wrong place to do it

mayasaxena commented on lucien-ios/Dashboard/DashboardComicCells/CollectionViewCell.swift:
```diff
@@ -0,0 +1,62 @@
+
+import UIKit
+
+final class CollectionViewCell: UICollectionViewCell {
```
Let's rename this to something more specific - `DashboardCollectionViewCell` would be better

mayasaxena commented on lucien-ios/Dashboard/DashboardComicCells/CollectionViewCell.swift:
```diff
@@ -0,0 +1,62 @@
+final class CollectionViewCell: UICollectionViewCell {
+
+    @IBOutlet private weak var dueDateLabel: UILabel!
+    @IBOutlet weak var ownerBorrowerNameLabel: UILabel!
```
Make IBOutlets private

mayasaxena commented on lucien-ios/Models/DashboardComic.swift:
```diff
@@ -0,0 +1,68 @@
+struct DashboardComic: Decodable {
+    var dashboardComic: Comic
+    var dashboardUser: User
+    var dashboardUserType: String
```
This should be a case of an enum, not a string

mayasaxena commented on lucien-ios/Dashboard/DashboardComicCells/CollectionViewCell.swift:
```diff
@@ -0,0 +1,62 @@
+    func configureLentOrBorrowedLabel(userType: String) {
+        switch userType {
+        case "borrower":
+            lentOrBorrowedLabel.text = "LENT TO"
```
This text should come from the configure method, instead of passing in the user type and determining it from there

mayasaxena commented on lucien-ios/Dashboard/DashboardComicCells/CollectionViewCell.swift:
```diff
@@ -0,0 +1,62 @@
+    // TO DO: load another image created by me if no comic book cover
+    func setImage(imageURL: String?) {
+        guard let url = URL(string: imageURL!) else { return }
+        let sampleImageData = try? Data(contentsOf: URL(string: "https://www.petsworld.in/blog/wp-content/uploads/2014/09/adorable-cat.jpg")!)
```
Should this URL be merged to develop?

mayasaxena commented on lucien-ios/Dashboard/DashboardComicCells/CollectionViewCell.swift:
```diff
@@ -0,0 +1,62 @@
+
+    // TO DO: load another image created by me if no comic book cover
+    func setImage(imageURL: String?) {
+        guard let url = URL(string: imageURL!) else { return }
```
Don't force unwrap

Commit:
PR changes and file renaming

mayasaxena commented on lucien-ios/Dashboard/DashboardCollectionViewModel.swift:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    func getComicDueDate(forIndex index: Int) -> Date {
+        guard let returnDate = dashboardComics[index].dashboardComic.returnDate else { return Date()}
```
Space after the `()`

mayasaxena commented on lucien-ios/Dashboard/DashboardComicCells/DashboardCollectionViewCell.swift:
```diff
@@ -0,0 +1,50 @@
+        guard let imageURL = imageURL else { return }
+        guard let url = URL(string: imageURL) else { return }
+        guard let data = try? Data(contentsOf: url) else {
+            return
```
This can be a one-liner since you're not doing anything but returning

mayasaxena reviewed (changes requested):
Couple of small changes + rebase onto develop

Commit:
Make Comic model protocol. Change DashboardComic to DashboardComicUser and create DashboardComic that conforms to Comic protocol

mayasaxena commented on lucien-ios/Models/Comic.swift:
```diff
@@ -8,133 +8,143 @@
-        case good
-        case fair
-        case poor
+enum Condition: Int {
```
Why did you move the enum out of the extension?

mayasaxena commented on lucien-ios/Models/Comic.swift:
```diff
@@ -8,133 +8,143 @@
+    var publisher: String? { get set }
+    var releaseDate: Date? { get set }
+    var comicPhotoURL: String? { get set }
+    var returnDate: Date? { get set }
```
This shouldn't be part of the protocol

mayasaxena commented on lucien-ios/Models/Comic.swift:
```diff
@@ -8,133 +8,143 @@
-struct Comic {
+protocol Comic: Decodable {
 
+    var comicID: String { get set }
```
This shouldn't be part of the protocol

mayasaxena commented on lucien-ios/Models/Comic.swift:
```diff
@@ -8,133 +8,143 @@
+protocol Comic: Decodable {
 
+    var comicID: String { get set }
+    var comicTitle: String { get set }
```
These values should probably be `get` only since it's better practice to create a new instance rather than mutate the current instance

IntrepidBuildServer commented on lucien-ios/AddAndEditNewComicBook/ComicFormViewModel.swift:
```diff
@@ -98,8 +98,14 @@ class ComicFormViewModel {
+        self.condition.value = Condition.convertStringToCondition(condition: comic.condition)
+    }
+
+
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 2. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Avertical_whitespace)

IntrepidBuildServer commented on lucien-ios/Models/AddEditComic.swift:
```diff
@@ -0,0 +1,71 @@
+}
+
+
+
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Avertical_whitespace)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Trailing Newline Violation: Files should have a single trailing newline. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Atrailing_newline)

IntrepidBuildServer commented on lucien-ios/Models/AddEditComic.swift:
```diff
@@ -0,0 +1,71 @@
+//  Copyright © 2017 Intrepid Pursuits. All rights reserved.
+//
+
+
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 2. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Avertical_whitespace)

IntrepidBuildServer commented on lucien-ios/Models/AddEditComic.swift:
```diff
@@ -0,0 +1,71 @@
+        case genre = "genre"
+        case releaseYear = "release_year"
+    }
+    
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Trailing Whitespace Violation: Lines should not have trailing whitespace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Atrailing_whitespace)

IntrepidBuildServer commented on lucien-ios/Models/AddEditComic.swift:
```diff
@@ -0,0 +1,71 @@
+    var condition: String?
+    var genre: String?
+    var releaseYear: String?
+    
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Trailing Whitespace Violation: Lines should not have trailing whitespace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Atrailing_whitespace)

IntrepidBuildServer commented on lucien-ios/Models/AddEditComic.swift:
```diff
@@ -0,0 +1,71 @@
+import Foundation
+
+struct AddEditComic: Comic, Codable {
+    
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Trailing Whitespace Violation: Lines should not have trailing whitespace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Atrailing_whitespace)

ghost commented:
SonarQube analysis reported 9 issues
* ![MINOR][MINOR] 9 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [ComicFormViewModel.swift#L85](https://github.com/IntrepidPursuits/lucien-ios/blob/2c5bdebf65762ea13408606b8eb624e61311f1d4/lucien-ios/AddAndEditNewComicBook/ComicFormViewModel.swift#L85): Trailing Whitespace Violation: Lines should not have trailing whitespace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Atrailing_whitespace)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

ghost commented:
<!--
  0 Errors
  1 Warning: Please provide a summary in th...
  2 Messages: Executed 2 tests, with 0 failu...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the Pull Request description.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.365 (0.367) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 1 test, with 0 failures (0 unexpected) in 8.885 (8.886) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>
